modules/slms_3/1.0/slms_3.smk

```diff
@@ -0,0 +1,591 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Laura Hilton
+# Module Author:    Laura Hilton
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package oncopipe and inspect packages. 
+import oncopipe as op
+import inspect
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Use modified setup_module function to skip the global `CFG` check. This module assigns the module config to a variable called `CFG_SLMS3` so that the submodules can still use `CFG`. 
```

Comment:
Did (or can) you document what you modified in this setup_module function? In the current form it's not clear to me what necessitates this duplication of code.

Reply:
Now that I'm thinking again about the reason I did this it might not be necessary. Will test and either remove this or add more detail to the comment.

Reply:
If it was just to avoid clobbering the CFG from another module I think I had a way around this that worked in the Sequenza or Battenberg modules.

Reply:
I think this was giving me grief in an earlier prototype when I tried to load the sub-modules before setting up this variable. It did turn out to be unnecessary to duplicate this function so it's gone now!